src/ring_theory/finiteness.lean

```diff
@@ -620,7 +791,12 @@ begin
 end
 
 instance ft_of_fg [comm_ring R] [monoid.fg M] : finite_type R (monoid_algebra R M) :=
-add_monoid_algebra.ft_of_fg.equiv (monoid_algebra.to_additive_alg_equiv R M).symm
+add_monoid_algebra.ft_of_fg.equiv (to_additive_alg_equiv R M).symm
+
+lemma ft_iff_fg [comm_ring R] [nontrivial R] :
```

Comment:
Should one direction of this be an instance as `ft_of_fg`?

Reply:
Yes, I think the forward direction is a good instance. The other direction can also be a lemma that assumes the right hand side as TC argument with `[..]`.